core/src/main/java/org/elasticsearch/index/mapper/internal/ParentFieldMapper.java

```diff
@@ -88,17 +79,13 @@
 
         private String parentType;
 
-        protected String indexName;
-
         private final String documentType;
 
         private final MappedFieldType parentJoinFieldType = Defaults.JOIN_FIELD_TYPE.clone();
 
-        private final MappedFieldType childJoinFieldType = Defaults.JOIN_FIELD_TYPE.clone();
-
         public Builder(String documentType) {
-            super(Defaults.NAME, Defaults.FIELD_TYPE, Defaults.FIELD_TYPE);
-            this.indexName = name;
+            super(Defaults.NAME, new ParentFieldType(Defaults.JOIN_FIELD_TYPE, documentType), Defaults.JOIN_FIELD_TYPE);
+            docValues(true);
```

Comment:
this is because `FieldMapper.Builder#docValuesSet` boolean field. It needs to be set to true, otherwise `#setupFieldType(...)` doesn't enable doc values on fieldType.

Maybe instead of invoking this `docValues(...)` method, the `FieldMapper.Builder` constructor should initialise `docValuesSet` value to whether doc values has been enabled on `fieldType` parameter?